### ISSUE TITLE
Incorrect typings warn + phantom variable/type name

### ISSUE THREAD

**Issue by lifeart** [JSMonk/hegel#126]
ref https://twitter.com/rage_monk/status/1250109296999829504?s=20

input
```js
let a = 12;
function b(n, m) {
    return n.toString() + m.toString();
}
let c = b(a, a);
```

output:

![image](https://user-images.githubusercontent.com/1360552/79253668-41a3da80-7e8c-11ea-8603-13a1c33bd0ed.png)

![image](https://user-images.githubusercontent.com/1360552/79253679-46688e80-7e8c-11ea-89b4-190b0c8c144c.png)

**Comment by thecotne** [comment]
this is interesting!

in flow and hegel (as i know) you can't have interface that primitive types can implement

for example in typescript you can write this

```
function b<T extends { toString(): string}>(n: T, m: T): string {
    return n.toString() + m.toString()
}

void b(1, 1);
```

describing generic with toString that returns string and number can be passed here because typescript does not care about number bean primitive and not extending object

but in flow and hegel you can't define interface for primitive types because those are nominally typed flow solves this by typing function by usage but. hegel has a design goal to no do that

probably correct solution would be to allow all values to implement interfaces so typescript style generic can be created

**Comment by JSMonk** [comment]
Fixed in `0.0.43` release. Thank you for the contribution ^_^.
